### ISSUE TITLE
Is there a difference between split() and split(" ") ?

### ISSUE THREAD

**Issue by MagLoe1** [Aequivinius/uibk-python#187]
Hello,

I just finished this week's exercise and it took me quite some time to figure out what my mistake was. All the palindromes were spotted correctly, but I still got an error message (see below). In the end, all tests passed when i changed split() to split(" "), but I still do not understand why there is a difference, as the default seperator for the split()-method is a space-character anyway. 

Thank you in advance and have a nice weekend,
Magdalena

The error message looked something like this:
...
assert palindrome.tokenize("")
AssertionError: assert[]
+ where []  = <function tokenize at 0x7f24c2e15dc0>(' ')
+   where <function tokenize at 0x7f24c2e15dc0> = palindrome.tokenize

**Comment by Aequivinius** [comment]
TLDR: `split()` and `split(" ")` are pretty much the same, but there's a mistake in the test. Please use `split(" ")`.

Long answer: `split()` splits on any type of white space (also tabs, for example), and returns an empty `list []` if run on the empty `string ""`. `"".split(" ")`, however, returns a `list` containing an empty `string` (`[ "" ]`). The test `assert tokenize("")` is a mistake, and it implicitly tests if `tokenize("")` returns a non-empty `list`, so it only passes for `split(" "). I've changed the test both `split()` and `split(" ")` should now work, but if not, please use `split(" ")` and accept my apologies for the mistake.

**Comment by MagLoe1** [comment]
Hello,
thank you for your answer, now everything is clear to me, I was just confused if there was a crucial difference that I had not thought of. :-)
Best regards
Magdalena
